Plugins/ScriptingPython/scriptingpython.cpp

```diff
@@ -582,15 +582,25 @@ SqlQueryPtr ScriptingPython::dbCommonEval(PyObject* sqlArg, const char* fnName)
 QVariant ScriptingPython::getVariable(const QString& name)
 {
     PyThreadState* state = PyThreadState_Get();
-    if (!state->frame)
+#if PY_VERSION_HEX < 0x03090000
+    PyFrameObject* frame = state->frame;
+#else
+    PyFrameObject* frame = PyThreadState_GetFrame(state);
+#endif
+    if (!frame)
         return QVariant();
 
     const char* varName = name.toUtf8().constData();
     PyObject* obj = nullptr;
 
-    PyFrame_FastToLocals(state->frame);
-    PyObject* locals = state->frame->f_locals;
-    PyObject* globals = state->frame->f_globals;
+    PyFrame_FastToLocals(frame);
+#if PY_VERSION_HEX < 0x03090000
+    PyObject* locals = frame->f_locals;
+    PyObject* globals = frame->f_globals;
+#else
+    PyObject* locals = PyFrame_GetLocals(frame);
+    PyObject* globals = PyFrame_GetGlobals(frame);
```

Comment:
Turns out `PyFrame_GetLocals` and `PyFrame_GetGlobals` are new in 3.11.

Reply:
(I'll fix it, I'm just letting you know)

Reply:
Ok. Thanks ;)